File: src/main/java/uk/co/lukestevens/utils/Wrapper.java
package uk.co.lukestevens.utils;

import java.util.Objects;

/**
 * A wrapper class for allowing objects to be set within lambda functions
 * 
 * @author luke.stevens
 *
 * @param <T> The type of object that this wraps
 */
public class Wrapper<T> {
	
	private T value;
	
	/**
	 * Creates a new wrapper without a value
	 */
	public Wrapper(){
		this(null);
	}
	
	/**
	 * Creates a wrapper with an initial value
	 * @param value The initial value of the wrapped object
	 */
	public Wrapper(T value){
		this.value = value;
	}
	
	/**
	 * @return The wrapped instance
	 */
	public T get() {
		return value;
	}

	/**
	 * Set the wrapped instance
	 * @param value
	 */
	public void set(T value) {
		this.value = value;
	}
	
	/**
	 * @return true if the wrapped instance is null
	 */
	public boolean isNull() {
		return value == null;
	}
	
	@Override
	public String toString() {
		return value == null? "null" : value.toString();
	}
	
	@Override
	public int hashCode() {
		return value == null? 0 : value.hashCode();
	}
	
	@Override
	public boolean equals(Object obj) {
		return Objects.equals(obj, value);
	}
}

File: src/main/kotlin/uk/co/lukestevens/utils/Wrapper.kt
package uk.co.lukestevens.utils

import java.util.Objects

/**
 * A wrapper class for allowing objects to be set within lambda functions
 *
 * @author luke.stevens
 *
 * @param <T> The type of object that this wraps
 */
class Wrapper<T>
/**
 * Creates a wrapper with an initial value
 * @param value The initial value of the wrapped object
 */
constructor(var value: T? = null) {

	/**
	 * @return true if the wrapped instance is null
	 */
	fun isNull(): Boolean {
		return value == null
	}

	/**
	 * @return The wrapped instance
	 */
	fun get(): T? {
		return value
	}

	/**
	 * Set the wrapped instance
	 * @param value
	 */
	fun set(value: T?) {
		this.value = value
	}

	@Override
	override fun toString(): String {
		return value?.toString()?: "null"
	}

	@Override
	override fun hashCode(): Int {
		return value?.hashCode()?: 0
	}

	@Override
	override fun equals(other: Any?): Boolean {
		return Objects.equals(other, value)
	}
}